lib/screens/log.ex

```diff
@@ -0,0 +1,20 @@
+defmodule Screens.Log do
+  @moduledoc "Conveniences for logging to Splunk and Sentry."
+
+  require Logger
+
+  def error(tag, data \\ []), do: log(:error, tag, data)
+  def warning(tag, data \\ []), do: log(:warning, tag, data)
+
+  @spec log(:error | :warning, String.t(), Enumerable.t({term(), term()})) :: :ok
+  defp log(level, tag, data) do
+    Logger.log(
+      level,
+      tag <> " " <> Enum.map_join(data, " ", fn {key, value} -> "#{key}=#{inspect(value)}" end)
```

Comment:
Probably unlikely, but technically anything can be passed in as a `key` right? Should we wrap it in an `inspect()` to be safe or maybe tighten the typespec to enforce that a keyword list is passed in?

Reply:
You can also pass in a Map, or anything that implements Enumerable by returning `{key, value}` tuples — there are existing logs that use maps, so I wanted to preserve it. Though it's a good point that this will not accept literally anything, I should correct it to `Enumerable.t({String.Chars.t(), term()})`.

To be clear there is no real "type safety" here — `Enumerable.t(_anything)` is identical to `Enumerable.t()`, which is an alias to `term()`. Protocols are an Elixir invention so there is no way to express "value that implements protocol X" in Dialyzer's type system. The value is solely in the documentation of programmer intent.